File: Core-backend-layer/Programming-fundamentals/Java/DesignPatterns/src/factoryMethod/ElectronicsStore.java
package factoryMethod;

public class ElectronicsStore extends Store {
    public ElectronicsStore(String storeName) {
        super(storeName);
    }

    @Override
    protected Worker createWorker(String workerName) {
        // The factory method for an electronics store creates salespersons
        return new Salesperson(workerName);
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/DesignPatterns/src/factoryMethod/electronics_store.py
from factoryMethod.store import Store
from factoryMethod.salesperson import Salesperson

class ElectronicsStore(Store):
    def __init__(self, store_name):
        super().__init__(store_name)

    def create_worker(self, worker_name):
        # The factory method for an electronics store creates salespersons
        return Salesperson(worker_name)